pyproject.toml

```diff
@@ -22,7 +22,8 @@ dynamic = ["version"]
 dependencies = [
     "requests>=2.31.0,<3",
     "packaging",
-    "typing-extensions"
+    "typing-extensions",
+    "setuptools-scm;python_version>='3.12'",
```

Comment:
`setuptools-scm` should only be required for the build group. It is used to derive the version information for Git